api/python/arrus/utils/probe_check.py

```diff
@@ -558,75 +683,134 @@ def validate(
 EXTRACTORS = dict([(e.feature, e) for e in
                    [MaxAmplitudeExtractor,
                     SignalDurationTimeExtractor,
-                    EnergyExtractor]])
+                    EnergyExtractor,
+                    FootprintSimilarityPCCExtractor]])
 
 
 class ProbeHealthVerifier:
     """
     Probe health verifier class.
     """
 
+    def get_footprint(
+            self,
+            cfg_path: str,
+            n: int,
+    )-> Footprint:
+        """
+        Creates and returns Footprint object.
+        """
+
+        rfs, metadata, masked_elements = self._acquire_rf_data(cfg_path, n)
+        footprint = Footprint(
+            rf=rfs,
+            metadata=metadata,
+            masked_elements=masked_elements,
+            creation_time=time.time(),
+
+        )
+        return footprint
+
     def check_probe(
             self,
-            cfg_path: str, n: int,
+            cfg_path: str,
+            n: int,
+            tx_frequency: float,
             features: List[FeatureDescriptor],
-            validator: ProbeElementValidator) -> ProbeHealthReport:
+            validator: ProbeElementValidator,
+            voltage: int=5,
+            footprint: Footprint=None,
```

Comment:
In my opinion the `footprint` parameter should be a part of the `FeatureDescriptor` class instead. My recommendation:
- add attribute `params: dict = dataclasses.field(default_factory=dict)` to the `FeatureDescriptor` class,
- for the  FootprintSimilarityPCCExtractor, it will be expected to provide params["footprint"] = array
- pass the feature params to the extractor constructor, e.g.
```
extractor = EXTRACTORS[feature.name](**feature.params)
```
- create an appropriate constructor for `FootPrintSimilarityPCCExtractor` (copy the footprint to the class attribute),
- in the `check_probe.py`, use  `FootprintSimilarityPCCExtractor` only when the footprint array is provided
- describe available features in the dosctring of `ProbeHealthVerified.check_probe`

Reply:
Any opinion on that?

Reply:
@zkLog Any opinion on that?

Reply:
@zkLog to decide, how much time it will take

Reply:
This can take quite a long time, at least for me. If this is a better solution than the current one, then let's come back to it in the future.